protoc-gen-gogo/generator/generator.go

```diff
@@ -1428,6 +1428,10 @@ func (g *Generator) generateEnum(enum *EnumDescriptor) {
 		g.PrintComments(fmt.Sprintf("%s,%d,%d", enum.path, enumValuePath, i))
 
 		name := ccPrefix + *e.Name
+		if gogoproto.IsEnumCustomName(e) {
```

Comment:
I prefer an error or warning.  Especially since we are going to stealthily apply this extension.  The user should know when he is specifying an extension that will have no effect.